pkg/logutil/merge_logger.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package logutil includes utilities to faciliate logging.
+package logutil
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/coreos/etcd/Godeps/_workspace/src/github.com/coreos/pkg/capnslog"
+)
+
+var normalPeriod = time.Second
+
+// line represents a log line that can be printed out
+// through capnslog.PackageLogger.
+type line struct {
+	level capnslog.LogLevel
+	str   string
+}
+
+func (l line) attach(s string) line {
+	return line{
+		level: l.level,
+		str:   l.str + " " + s,
+	}
+}
+
+type status struct {
+	period time.Duration
+
+	mu    sync.Mutex // protect fields below
+	start time.Time  // time to start merging
+	count int        // number of lines from start field
+}
+
+func (s *status) canPrint(now time.Time) bool {
```

Comment:
I feel we should change this to `isMerging`. First it makes the code logic focus on `merge`. Second the code block will be more readable like 

```go
if s.isMerging(){
    continue
}
``` 

instead of

```go
if !s.canPrint() {
    continue
}
```